packages/by-name/cloud-api-adaptor/package.nix

```
@@ -86,6 +90,21 @@ buildGoModule rec {
         "SC2153"
       ];
     };
+
+    setup-nat-for-imds = writeShellApplication {
+      name = "setup-nat-for-imds";
+      runtimeInputs = [
+        iproute2
+        iptables
+        sysctl
+        gawk
+      ];
+      text = builtins.readFile "${cloud-api-adaptor.src}/src/cloud-api-adaptor/podvm/files/usr/local/bin/setup-nat-for-imds.sh";
```

```suggestion
      # TODO(burgerdev): generalize for all link-local IPs and investigate routing simplification 
      text = builtins.readFile "${cloud-api-adaptor.src}/src/cloud-api-adaptor/podvm/files/usr/local/bin/setup-nat-for-imds.sh";
```
This script is a bit weird: I wonder if a route `via $host` and the masquerade would be enough. We should reiterate after we can test it.

---

My networking knowledge does not suffice to tell from your comment what you want to do, but I'm happy to add this comment :D